crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -851,18 +876,44 @@ impl<TX: DbTx, N: NodeTypes> AccountExtReader for DatabaseProvider<TX, N> {
         &self,
         range: RangeInclusive<BlockNumber>,
     ) -> ProviderResult<BTreeMap<Address, Vec<u64>>> {
-        let mut changeset_cursor = self.tx.cursor_read::<tables::AccountChangeSets>()?;
+        tracing::trace!(target: "provider::static_file", ?range, "Getting changesets for range");
+        let highest_static_block = self
+            .static_file_provider
+            .get_highest_static_file_block(StaticFileSegment::AccountChangeSets);
+
+        if let Some(highest) = highest_static_block {
```

Comment:
is it possible to move this logic by impl `AccountExtReader` for `StaticFileProvider` similarly to the others ?